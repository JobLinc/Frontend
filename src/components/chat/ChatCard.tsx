import React, { useEffect, useState } from "react";
import { EllipsisVertical } from "lucide-react";
import Checkbox from "./UI/CheckBox";
import { ChatCardProps } from "./interfaces/Chat.interfaces";

export default function ChatCard({
  chatId,
  chatPicture,
  chatName,
  lastMessage,
  sentDate,
  onClick,
  unseenCount,
  isRead,
  className = "w-full",
}: ChatCardProps) {
  const [isMenuOpen, setIsMenuOpen] = useState(false);

  const toggleMenu = (e: React.MouseEvent) => {
    e.stopPropagation();
    setIsMenuOpen((prev) => !prev);
  };

  const handleDelete = () => {
    console.log(`Deleting chat: ${chatId}`);
    setIsMenuOpen(false);
  };

  const handleMarkReadUnread = () => {
    console.log(`Toggling read/unread for: ${chatId}`);
    setIsMenuOpen(false);
  };

  return (
    <div
      onClick={onClick}
      id={chatId}
      className={`${className} ${!isRead ? "bg-SoftRed hover:bg-hoverSoftRed" : "bg-charcoalWhite hover:bg-gray-200"}  
  flex items-center p-3 cursor-pointer relative rounded-lg shadow-sm group`}
    >
      <div className="relative w-12 h-12 shrink-0 group">
        <img
          className="rounded-full w-full h-full object-cover"
          alt={chatName}
          src={chatPicture}
        />
        <div
          className="absolute inset-0 flex items-center justify-center bg-darkGray bg-opacity-50
      rounded-full opacity-0 group-hover:opacity-100 transition-all"
        >
          <Checkbox />
        </div>
      </div>

      <div className="flex flex-1 justify-between items-center ml-3 overflow-hidden">
        <div className="flex flex-col overflow-hidden">
          <div className="font-medium">{chatName}</div>
          <div className="text-sm text-gray-500 truncate max-w-[250px]">
            {lastMessage}
          </div>
        </div>

        <div className="flex items-center gap-2 min-w-[60px] justify-end">
          <span className="text-xs text-gray-400">{sentDate}</span>
          <button
            onClick={toggleMenu}
            className={`p-2 rounded-full z-20 transition
          ${!isRead ? "hover:bg-hoverDarkBurgundy" : "hover:bg-gray-300"}`}
          >
            <EllipsisVertical className="w-5 h-5 text-gray-600 dark:text-gray-300 opacity-0 group-hover:opacity-100" />
          </button>
          {isMenuOpen && (
            <div className="absolute top-12 right-3 w-36 bg-white shadow-md rounded-md p-2 z-50">
              <button
                className="w-full text-left px-3 py-1 text-sm hover:bg-gray-200"
                onClick={handleDelete}
              >
                Delete
              </button>
              <button
                className="w-full text-left px-3 py-1 text-sm hover:bg-gray-200"
                onClick={handleMarkReadUnread}
              >
                {isRead ? "Mark as Unread" : "Mark as Read"}
              </button>
            </div>
          )}

          {unseenCount > 0 && (
            <div
              className="absolute top-2 right-3 text-xs text-white bg-red-500 rounded-full px-1 
          opacity-100 group-hover:opacity-0"
            >
              {unseenCount}
            </div>
          )}
        </div>
      </div>
    </div>
  );
}
