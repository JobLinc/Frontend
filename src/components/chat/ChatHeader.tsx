import React from "react";
import { MoreHorizontal, X } from "lucide-react";

interface User {
  name: string;
  profilePicture: string;
  status: "online" | "offline";
}

interface ChatHeaderProps {
  onClick?: () => void;
  user: User;
  onClose: () => void;
}

function ChatHeader({ onClick, user, onClose }: ChatHeaderProps) {
  return (
    <div
      className={`flex items-center justify-between bg-charcoalWhite 
        p-2 dark:text-charcoalBlack rounded-t-lg border-b-1 z-10
        border-gray-300 cursor-pointer h-[60px] `}
      onClick={onClick}
    >
      <div className="flex items-center space-x-3">
        <div className="relative">
          <img
            src={user.profilePicture}
            alt={user.name}
            className="w-10 h-10 rounded-full"
          />

          <span
            className={`absolute bottom-0 right-0 w-3 h-3 rounded-full border-2 ${
              user.status === "online" ? "bg-green-500" : "bg-mutedSilver"
            }`}
          />
        </div>

        <div>
          <p className="text-lg font-semibold">{user.name}</p>
          <p className="text-sm text-mutedSilver">
            {user.status === "online" ? "Available" : "Offline"}
          </p>
        </div>
      </div>

      <div className="flex items-center space-x-2 ">
        <button className="text-mutedSilver hover:text-warmBlack">
          <MoreHorizontal size={20} />
        </button>
        <button
          onClick={onClose}
          className="text-mutedSilver hover:text-warmBlack"
        >
          <X size={20} />
        </button>
      </div>
    </div>
  );
}

export default ChatHeader;
