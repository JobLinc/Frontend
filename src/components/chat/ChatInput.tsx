import { useState, useEffect } from "react";
import { Paperclip, Image, Smile } from "lucide-react";
import { ChatInputProps } from "./interfaces/Chat.interfaces";

function ChatInput({ chatId, onSendMessage, onTypingMessage }: ChatInputProps) {
  const [message, setMessage] = useState("");
  const [isFocused, setIsFocused] = useState(false);
  const [isTyping, setIsTyping] = useState(false);

  useEffect(() => {
    console.log("typing happnes", isTyping);
    onTypingMessage(isTyping);
  }, [isTyping]);
  useEffect(() => {
    if (!message.trim()) setIsTyping(false);
    else setIsTyping(true);
  }, [message]);

  const handleSendMessage = () => {
    if (!message.trim() || !chatId) return;
    onSendMessage(message);
    setMessage("");
    onTypingMessage(false);
  };

  return (
    <div
      className="flex flex-col items-center p-3
     bg-charcoalWhite dark:bg-warmBlack border-t border-gray-200 relative"
    >
      <textarea
        name="msg"
        value={message}
        onChange={(e) => setMessage(e.target.value)}
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
        className={`flex-1 w-full bg-warmWhite dark:bg-darkGray text-charcoalBlack
         dark:text-charcoalWhite p-2 rounded-lg resize-none h-10 focus:outline-none
         shadow-inner transition-all ${
           isFocused ? "ring-2 ring-crimsonRed" : "ring-1 ring-gray-400"
         }`}
        placeholder="Write a message..."
      />

      <div className="flex justify-between w-full pt-4">
        <div>
          <button
            className="text-mutedSilver dark:text-charcoalWhite hover:text-charcoalBlack 
            dark:hover:text-warmWhite p-2"
          >
            <Paperclip size={20} />
          </button>
          <button
            className="text-mutedSilver dark:text-charcoalWhite hover:text-charcoalBlack 
            dark:hover:text-warmWhite p-2"
          >
            <Image size={20} />
          </button>
          <button
            className="text-mutedSilver dark:text-charcoalWhite hover:text-charcoalBlack 
            dark:hover:text-warmWhite p-2"
          >
            <Smile size={20} />
          </button>
        </div>

        <button
          id="send-msg-btn"
          onClick={handleSendMessage}
          disabled={!message.trim()}
          className={`ml-2 p-1 rounded-l-full font-bold rounded-r-full transition-colors
            ${
              message.trim()
                ? "bg-crimsonRed hover:bg-darkBurgundy text-warmWhite"
                : "bg-mutedSilver text-charcoalBlack cursor-not-allowed"
            } w-15 md:w-20`}
        >
          Send
        </button>
      </div>
    </div>
  );
}

export default ChatInput;
