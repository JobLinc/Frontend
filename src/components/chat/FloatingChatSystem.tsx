import FloatingChatSidebar from "./FloatingChatSidebar";
import FloatingChatWindow from "./FloatingChatWindow";
import useChats from "../../hooks/useChats";
import { ChatIdProvider } from "../../context/ChatIdProvider";

function FloatingChatSystem() {
  const { opnedChatsId } = useChats();
  return (
    <>
      <FloatingChatSidebar />
      {opnedChatsId.map((opnedChatId, index) => {
        return (
          <ChatIdProvider id={opnedChatId}>
            <FloatingChatWindow
              key={opnedChatId}
              style={{ right: `${index * 410 + 330}px` }}
              chatId={opnedChatId}
            />
          </ChatIdProvider>
        );
      })}
    </>
  );
}

export default FloatingChatSystem;
