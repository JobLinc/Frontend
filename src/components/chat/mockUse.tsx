import { createContext, useContext, useRef, ReactNode } from "react";

interface UserContextType {
  user: string;
}

const UserContext = createContext<UserContextType | undefined>(undefined);

export function UserProvider({
  children,
  userId,
}: {
  children: ReactNode;
  userId: string;
}) {
  const userRef = useRef<string>(userId);
  const user = userRef.current;

  return (
    <UserContext.Provider value={{ user }}>{children}</UserContext.Provider>
  );
}

export const useUser = () => {
  const context = useContext(UserContext);
  if (!context) {
    throw new Error("useUser must be used within a UserProvider");
  }
  return context;
};
