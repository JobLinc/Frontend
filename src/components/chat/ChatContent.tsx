import { memo, useState, useEffect, useRef } from "react";
import ChatMessages from "./ChatMessages";
import ChatInput from "./ChatInput";
import {
  subscribeToMessages,
  unsubscribeFromMessages,
  sendMessage,
  typing,
  stopTyping,
  listenToOpenChatErrors,
} from "@services/api/ChatSocket";
import { RecievedMessage } from "./interfaces/Message.interfaces";
import { User } from "./interfaces/User.interfaces";
import useChats from "@hooks/useChats";
import { fetchChatData, createChat } from "@services/api/chatServices";
import useChatid from "@context/ChatIdProvider";
import useNetworkUserId from "@context/NetworkUserIdProvider";
import UserTypingIndicator from "./UserTyping";
import GroupChatSetting from "./GroupSetting/GroupChatSetting";
import SeenBy from "./SeenBy";
function ChatContent({ className }: { className?: string }) {
  const [users, setUsers] = useState<User[]>([]);
  const [messages, setMessages] = useState<RecievedMessage[]>([]);
  const [loading, setLoading] = useState(true);

  const { chatId, setChatId } = useChatid();
  const { setOpnedChats } = useChats();
  const { usersId } = useNetworkUserId();
  const userIdRef = useRef<string | null>(localStorage.getItem("userId"));
  const messagesEndRef = useRef<HTMLDivElement | null>(null);
  const [typingUsers, setTypingUsers] = useState<string[]>([]);
  const chatType = useRef<string>("");
  let messageDelivered = useRef<boolean>(false);

  const handleMessageStatus = () => {
    setTimeout(() => {
      if (!messageDelivered.current) {
        messageDelivered.current = true;
        setMessages((prevMsgs) =>
          prevMsgs.map((msg) =>
            msg.messageId === messageId ? { ...msg, status: "failed" } : msg,
          ),
        );
      }
    }, 1000);
  };

  useEffect(() => {
    messagesEndRef.current?.scrollIntoView({ behavior: "smooth" });
  }, [messages]);

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);
      if (usersId.length && !chatId) {
        console.log("Create new chat with: ", usersId);
        const data = await createChat(usersId);
        setUsers(data.participants);
        console.log(data.participants);
        setMessages(data.messages);
        setChatId(data.chatId);
        setOpnedChats((prev) => [...prev]);
        chatType.current = data.chatType;
      } else {
        const data = await fetchChatData(chatId);
        setUsers(data.participants);
        setMessages(data.messages);
        chatType.current = data.chatType;
      }
      setLoading(false);
    };

    fetchData();
    subscribeToMessages(
      chatId,
      (message) => setMessages((prev) => [...prev, message]),
      (userId) =>
        setMessages((prev) =>
          prev.map((msg) => ({ ...msg, seenBy: [...msg.seenBy, userId] })),
        ),
      (userId) =>
        setTypingUsers((prevTypingUsers) => {
          if (!prevTypingUsers.includes(userId)) {
            return [...prevTypingUsers, userId];
          }
          return prevTypingUsers;
        }),
      (userId) =>
        setTypingUsers((prev) => prev.filter((preUser) => preUser != userId)),
    );
    listenToOpenChatErrors();
    return () => {
      unsubscribeFromMessages(chatId);
    };
  }, [chatId]);

  const messageId = Date.now().toString();

  const handleSendMessage = async (
    message: string | File | string,
    type: string,
  ) => {
    const newMessage: any = {
      messageId,
      senderId: userIdRef.current,
      time: new Date(),
      seenBy: [],
      status: "sent",
      content: {},
    };

    switch (type) {
      case "text":
        newMessage.content.text = message;
        break;
      case "image":
        newMessage.content.image = message;
        break;
      case "video":
        newMessage.content.video = message;
        break;
      case "document":
        newMessage.content.document = message;
        break;
      default:
        console.warn("Unknown message type:", type);
        newMessage.content.text = typeof message === "string" ? message : "";
        break;
    }

    setMessages((prevMsgs) => [...prevMsgs, newMessage]);

    if (chatId) {
      handleMessageStatus();
      sendMessage(chatId, newMessage, () => {
        messageDelivered.current = true;
        setMessages((prevMsgs) =>
          prevMsgs.map((msg) =>
            msg.messageId === messageId ? { ...msg, status: "delivered" } : msg,
          ),
        );
      });
    }
  };

  const handleTypingMessage = (isTyping: boolean) => {
    if (!chatId) return;
    switch (isTyping) {
      case true:
        typing(chatId);
        break;
      case false:
        stopTyping(chatId);
        break;
    }
  };

  return (
    <>
      {chatType.current === "group" && (
        <GroupChatSetting users={users} setUsers={setUsers} chatId={chatId} />
      )}
      <div className={`${className} flex flex-col flex-1 overflow-y-hidden`}>
        <div className="h-full overflow-y-auto bg-gray-100">
          <ChatMessages users={users} messages={messages} loading={loading} />
          {typingUsers.map((typingUserId) => (
            <UserTypingIndicator
              key={typingUserId}
              userImage={
                users.find((user) => user.userId == typingUserId)
                  ?.profilePicture
              }
            />
          ))}
          <div ref={messagesEndRef} />
        </div>
        <ChatInput
          chatId={chatId}
          onSendMessage={handleSendMessage}
          onTypingMessage={handleTypingMessage}
          className=""
        />
      </div>
    </>
  );
}

export default memo(ChatContent);
