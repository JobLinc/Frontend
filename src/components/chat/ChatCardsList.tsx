import React, { useEffect, useState, memo } from "react";
import ChatCard from "./ChatCard";
import { fetchChats } from "../../services/api/api";
import { connectChatSocket } from "../../services/api/socket";
import { useUser } from "./mockUse";
import { ChatCardInterface } from "./interfaces/Chat.interfaces";

const ChatCardsList = ({
  onCardClick,
}: {
  onCardClick: (id: string) => void;
}) => {
  const [chats, setChats] = useState<ChatCardInterface[]>([]);
  const chatSocket = connectChatSocket();
  const { user } = useUser();
  console.log("----------------ChatCardsList----------------");
  useEffect(() => {
    const fetchData = async () => {
      try {
        if (!user) return;
        const data = await fetchChats(user);
        console.log("Data:  ", data);
        setChats(data);
      } catch (error) {
        console.error("Error fetching chat data:", error);
      }
    };
    fetchData();

    chatSocket.on("ChatsUpdate", (newChat: ChatCardInterface) => {
      setChats((prev) => [...prev, newChat]);
    });

    return () => {
      chatSocket.off("ChatsUpdate");
    };
  }, [user]);
  console.log("Chats:", chats);
  return (
    <div className="mt-4">
      {chats.length > 0 ? (
        chats.map((chatCard) => (
          <ChatCard
            key={chatCard.chatId}
            chatId={chatCard.chatId}
            chatPicture={chatCard.chatPicture}
            chatName={chatCard.chatName}
            lastMessage={chatCard.lastMessage}
            sentDate={chatCard.sentDate}
            onClick={() => onCardClick(chatCard.chatId)}
          />
        ))
      ) : (
        <div className="text-gray-500 text-center mt-4">No chats found</div>
      )}
    </div>
  );
};

export default memo(ChatCardsList);
