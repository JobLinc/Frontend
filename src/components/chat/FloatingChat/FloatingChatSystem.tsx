import FloatingChatSidebar from "./FloatingChatSidebar";
import FloatingChatWindow from "./FloatingChatWindow";
import useChats from "../../../hooks/useChats";
import { ChatIdProvider } from "../../../context/ChatIdProvider";
import { useUser } from "../mockUse";
import { useEffect } from "react";

function FloatingChatSystem() {
  const { opnedChatsId } = useChats();
  const { setUser } = useUser();
  useEffect(() => {
    const UserId = window.prompt("Enter User ID:") || "1";
    setUser(UserId);
  }, []);
  return (
    <>
      <FloatingChatSidebar />
      {opnedChatsId.map((opnedChatId, index) => {
        return (
          <ChatIdProvider id={opnedChatId}>
            <FloatingChatWindow
              key={opnedChatId}
              style={{ right: `${index * 410 + 330}px` }}
            />
          </ChatIdProvider>
        );
      })}
    </>
  );
}

export default FloatingChatSystem;
