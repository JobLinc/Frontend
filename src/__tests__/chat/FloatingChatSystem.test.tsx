import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "../../../test/setupTests";
import ChatProvider from "../../context/ChatsIdProvider";
import FloatingChatSystem from "../../components/chat/FloatingChat/FloatingChatSystem";
import { UserProvider } from "../../components/chat/mockUse";
import { expect, it, describe } from "vitest";

describe("FloatingChatSystem \n", () => {
  it("Should RenderChatSystem", () => {
    const { container } = render(
      <UserProvider userId={"0"}>
        <ChatProvider>
          <FloatingChatSystem />
        </ChatProvider>
      </UserProvider>,
    );
    expect(container).toMatchSnapshot();
  });
  it("fetches and displays chat cards with correct data", async () => {
    render(
      <UserProvider userId={"4"}>
        <ChatProvider>
          <FloatingChatSystem />
        </ChatProvider>
      </UserProvider>,
    );

    await waitFor(() => {
      expect(screen.getByText("Mohaned")).toBeInTheDocument();
      expect(
        screen.getByText("Sounds good! See you then."),
      ).toBeInTheDocument();
      expect(screen.getByText("Michael")).toBeInTheDocument();
      expect(screen.getByText("Happy birthday bro!")).toBeInTheDocument();
    });
  });
  describe("Clicks on a certain chat card\n", () => {
    it("Renders Flaoting chat window", async () => {
      render(
        <UserProvider userId={"4"}>
          <ChatProvider>
            <FloatingChatSystem />
          </ChatProvider>
        </UserProvider>,
      );
      const mohanedCard = await screen.findByText("Mohaned");
      await userEvent.click(mohanedCard);
      expect(screen.getByTestId("test-floatingWindow")).toBeInTheDocument();
    });

    it("Should displays messages correctly", async () => {
      render(
        <UserProvider userId={"4"}>
          <ChatProvider>
            <FloatingChatSystem />
          </ChatProvider>
        </UserProvider>,
      );

      const mohanedCard = await screen.findByText("Mohaned");
      expect(mohanedCard).toBeInTheDocument();

      const michaelCard = await screen.findByText("Michael");
      expect(michaelCard).toBeInTheDocument();

      await userEvent.click(mohanedCard);

      const mohanedMessage = await screen.findByTestId("1-1633024800000");
      expect(mohanedMessage).toBeInTheDocument();
      expect(mohanedMessage).toHaveTextContent("Mohaned");
      expect(mohanedMessage).toHaveTextContent("Hey John!");

      const johnMessage = await screen.findByTestId("4-1633024800001");
      expect(johnMessage).toBeInTheDocument();
      expect(johnMessage).toHaveTextContent("John");
      expect(johnMessage).toHaveTextContent("Hey Mohaned!");
    });
  });
});
