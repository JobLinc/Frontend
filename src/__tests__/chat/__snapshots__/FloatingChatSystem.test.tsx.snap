// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FloatingChatSystem 
 > Should RenderChatSystem 1`] = `
<div>
  <div
    class="w-72 shadow-[0px_0px_10px_rgba(0,0,0,0.25)] 
                rounded-t-lg md:block hidden
                dark:bg-darkGray fixed bottom-0 right-0 mr-8
                cursor-pointer transition-transform duration-300
                translate-y-[calc(100%-56px)]"
  >
    <header
      class="flex w-full h-14 bg-charcoalWhite dark:bg-darkGray 
                   items-center gap-x-4 px-4 border-b-2 border-gray-200 
                   rounded-t-2xl "
    >
      <div
        class="flex items-center gap-x-4 w-full"
      >
        <img
          alt="User Avatar"
          class="rounded-full w-10 h-10 border border-mutedSilver"
          src="https://randomuser.me/api/portraits/men/2.jpg"
        />
        <div
          class="text-charcoalBlack dark:text-charcoalWhite font-semibold"
        >
          Messaging
        </div>
      </div>
    </header>
    <div
      class="flex items-center justify-center "
    >
      <div
        class="h-10 flex items-center overflow-hidden cursor-pointer"
      >
        <input
          class=" border-none outline-none text-sm caret-crimsonRed"
          id="input"
          name="text"
          placeholder="Search..."
          type="text"
        />
        <label
          class="cursor-text px-3"
          for="input"
        >
          <svg
            class="w-4 h-4"
            data-testid="test-search-icon"
            viewBox="0 0 512 512"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="rgb(114, 114, 114)"
            />
          </svg>
        </label>
        <div
          class="text-darkBurgundy ml-4"
          data-testid="test-filter-icon"
        >
          <svg
            class="h-8"
            fill="none"
            viewBox="0 0 30 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class=""
              d="M14 5C14 3.89543 13.1046 3 12 3C10.8954 3 10 3.89543 10 5C10 6.10457 10.8954 7 12 7C13.1046 7 14 6.10457 14 5ZM14 5H20M16 12C16 13.1046 16.8954 14 18 14C19.1046 14 20 13.1046 20 12C20 10.8954 19.1046 10 18 10C16.8954 10 16 10.8954 16 12ZM16 12H13M20 19H11M4 12H10M4 5L7 5M6 17C7.10457 17 8 17.8954 8 19C8 20.1046 7.10457 21 6 21C4.89543 21 4 20.1046 4 19C4 17.8954 4.89543 17 6 17Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="mt-4"
    >
      <div
        class="text-gray-500 text-center mt-4"
      >
        No chats found
      </div>
    </div>
  </div>
</div>
`;
